### DISCUSSION TITLE
Repeatable relationship field parent

### DISCUSSION THREAD

**Discussion by sagargc** [Laravel-Backpack/community-forum#667] [Q&A (Help)]
Hello everyone!!

I have my three models called page, block and content.
Page has multiple blocks and contents.

Page : Home , About Us and Products
Block: Text, Image_Text, Image_Video, Article
Content:  page_id, block_id, title, body, image, video, article as array.

I want to add multiple repetition of particular main block based on repeatable type of backpack features.

For example:

```php
CRUD::addField([
    'name'      => 'text_blocks',
    'label'     => 'Text Editor',
    'subfields' => [         
        [
            'name'  => 'title',
            'label' => "Title",
            'type'  => 'text'
        ],
        [
            'name'  => 'body',
            'label' => "Body",
            'type'  => 'text'
        ],
        [
            'name'  => 'image',
            'label' => "Image",
            'type'  => 'upload'
        ],
        [
            'name'  => 'video',
            'label' => "Title",
            'type'  => 'url'
        ],
        [
            'name'  => 'article',
            'label' => "Article",
            'type'  => 'text'
        ]
    ],        
    'wrapper'   => [
        'class' => 'form-group col-md-12 text repetable_block',
        'id' => 'repetable_block_text'
    ], 
    'new_item_label'  => 'Add Text Editor', 
    'sortable'   => true,
    'reorder' => 'order', 
]);
```

I want to repeat the text_blocks more than one times or any type of blocks need to repeat, and the subfields data  will be saved into coronets tables. Furthermore, this need to be auto load into edit mode as well.

My query is: how could I make subfields  main block repeatable?

Thank you for your time.

Kind regards,

Sagar

**Comment by promatik** [comment]
Hi @sagargc!
I read your comment 3 times, and I found it very difficult to understand, sorry 😶

But here is my opinion;
1) First, have you heard about [PageManager](https://github.com/Laravel-Backpack/PageManager)?
It is a package to deal with pages, it might make your life much easier, please take a look 👌

2) Also, when you want to deal with fields, fields depending on each other and having its behaviour changed during user interaction, for that we have [CrudField JavaScript Library](https://backpackforlaravel.com/docs/6.x/crud-fields-javascript-api), please take a look 👌

If you still need further help with your problem, please let me know, but please consider giving us a better description of your needs and the models relations (like, I didn't get if both block and content are child of page, of if content is child of block and block child of page(?)) and maybe an example, because I can't understand the differences between the block `Image_Text` and content `image`, why one is block and other is content(?) (just as example, because generally I didn't get your issue).

Regards 🙌

**Reply by sagargc** [reply]
Thank you @promatik for the answer.

Actually I wanted is that how can I add multiple repeatable container-repeatable-elements block by dynamic with same field_name. Likewise subfields, but I want create subfields' parent block repeatable by button click.

Thank you.
